lib/oppen/token.rb

```diff
@@ -39,6 +39,13 @@ def initialize(value, width: value.length)
       def to_s = value
     end
 
+    # Single whitespace token that is not printed if followed by a linebreak.
+    class Whitespace < ::Oppen::Token::String
+      def initialize
+        super(' ')
+      end
+    end
```

Comment:
Can't we work with a single whitespace token with many actual whitespaces in it? Does it make any sense? It will help reduce reallocations of the tokens stack when it grows bigger than its initial size.